snuba/snuba_migrations/generic_metrics/0058_distributions_raw_add_sampling_weight.py

```diff
@@ -0,0 +1,51 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    local_table_name = "generic_metric_distributions_raw_local"
+    dist_table_name = "generic_metric_distributions_raw_dist"
+    storage_set_key = StorageSetKey.GENERIC_METRICS_DISTRIBUTIONS
+
+    columns: Sequence[Column[MigrationModifiers]] = [
+        Column(
+            "sampling_weight",
+            UInt(64, MigrationModifiers(codecs=["ZSTD(1)"], default=str("1"))),
```

Comment:
Is this a UInt or a Float?

Reply:
Sampling weight is a `Uint`. This represents the multiple the metric should be scaled by. We decided to store this instead of the sampling factor. For example. if sampling factor is 1% or 0.1, then sampling weight is 100.

`sampling_weight = sampling factor^(-1)`